### DISCUSSION TITLE
Change autosave behavior

### DISCUSSION THREAD

**Discussion by aderici** [Budibase/budibase#6136] [Ideas]
Hi,

I have been using budibase for a while now and i think there are a lot of problems and unpredictability with trying to "magically autosave".

I believe it will provide much better developer experience if i can simply have a save button which i can press and get a feedback of successful save (button grays out when nothing to save, becomes active when unsaved changes, after press disables itself and re-enables after save event succeeds)

Currently automations are driving me crazy as they try to save with every keypress and mostly fail with 409 leaving a weird state being saved, my codes half way there of variable names half written. And the error message toast doesnt remove itself leaving me with 10s of toasts to press x and close.

Also within the designer i noticed data source filters are not saving most of the time and i have to press here and there to make it save successfully. 

All this trouble could probably be avoided by changing this to "press the button, get the reaction" type of behavior which i'd like to advocate here.

**Comment by melohagan** [comment]
Hi @aderici 

I can imagine that is frustrating, but I would consider this more of a bug if you're getting 409s. 

But maybe there are certain scenarios, such as automations were a manual save button would be more suitable.

**Reply by aderici** [reply]
Personally i’m advocating against this behavior. I’d still not feel comfortable if it magically worked because how can i be sure? And the system could nt keep its promise, increasing the trust barrier. 

A) the idea that every keypress can become an api call and its fine, without implementing a throttling / debouncing mechanism or involving websockets in the game is simply wrong

B) the design choice opens amazing amount of bug possibilities, also end user frustration (if they tend to think like me) so i still personally advocate against the feature, would like to have my save button and relevant indicators so i know whats happening without having to work with inspect window open all the time.

In general, Budibase got me from point a to point b really fast. That is both amazing and increasing the expectations as well as frustration when something does not perform the way you’d expect it to.
